File: medplum-java/medplum-model/src/main/java/com/fhir/Account_Guarantor.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.Boolean;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableAccount_Guarantor.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface Account_Guarantor {
  Reference party();

  Optional<Boolean> onHold();

  Optional<String> id();

  Optional<List<Extension>> extension();

  Optional<List<Extension>> modifierExtension();

  Optional<Period> period();

  static ImmutableAccount_Guarantor.PartyBuildStage builder() {
    return ImmutableAccount_Guarantor.builder();
  }
}

File: medplum-typescript/packages/sdk/src/model/fhir/Account_Guarantor.ts
import { Extension } from "./Extension";
import { Period } from "./Period";
import { Reference } from "./Reference";

export interface Account_Guarantor {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    party: Reference;
    onHold: boolean | null | undefined;
    period: Period | null | undefined;
}
